File: ddPoliglotV6.Data/Models/WordUser.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ddPoliglotV6.Data.Models
{
    public class WordUser
    {
        [Key]
        public int WordUserID { get; set; }

        public int WordID { get; set; }

        [ForeignKey("WordID")]
        public Word Word { get; set; }

        public Guid UserID { get; set; }

        public int LanguageID { get; set; }

        [ForeignKey("LanguageID")]
        public Language Language { get; set; }

        public int Grade { get; set; }

        public int LastRepeatWordPhraseId { get; set; }

        public int LastRepeatInLessonNum { get; set; }

        public int LastRepeatInArticleByParamID { get; set; }

        public string RepeatHistory { get; set; }

        public int SourceType { get; set; } // 0 - set grade by hand, 1- by auto (language level)
    }
}

File: ddpoliglot_v6_flutter/lib/models/data/WordUser.dart
// ignore_for_file: file_names
import 'Word.dart';
import './Language.dart';

import 'package:json_annotation/json_annotation.dart';

part 'WordUser.g.dart';

@JsonSerializable()
class WordUser {
  int wordUserID;
  int wordID;
  Word? word;
  String userID;
  int languageID;
  Language? language;
  int grade;
  int lastRepeatInArticleByParamID;
  int lastRepeatInLessonNum;
  int lastRepeatWordPhraseId;
  int sourceType; // 0 - set grade by hand, 1- by auto (language level)

  WordUser(
      this.wordUserID,
      this.wordID,
      this.word,
      this.userID,
      this.languageID,
      this.language,
      this.grade,
      this.lastRepeatInArticleByParamID,
      this.lastRepeatInLessonNum,
      this.lastRepeatWordPhraseId,
      this.sourceType);

  factory WordUser.fromJson(Map<String, dynamic> data) =>
      _$WordUserFromJson(data);

  Map<String, dynamic> toJson() => _$WordUserToJson(this);
}
